.github/workflows/ci.yml

```diff
@@ -0,0 +1,459 @@
+          # in practice there are still many sporadic "Could not authenticate against github.com" errors
+          # there's 1,000 requests per hour limit when using this token, likely it get exceeded
+          # fairly easily when using a fork with multiple jobs in a matrix
+          #env:
+          # COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Configure PHP
+        run: |
+          # Set memory limit and disable xdebug if not running phpcoverage
+          if [ -z $(which php) ]; then
+            echo "PHP not installed, skipping" && exit 0
+          fi
+
+          # github linux runners have 7GB of RAM
+          # https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+          # Set a high memory limit, particularly for php coverage tests
+          PHP_MEMORY_LIMIT=6144M
+          # Assign less memory for behat tests so that chrome has plenty of memory available
+          if [ "${{ matrix.endtoend }}" == "true" ]; then
+            PHP_MEMORY_LIMIT=4096M
+          fi
+          echo "PHP_MEMORY_LIMIT is $PHP_MEMORY_LIMIT"
+
+          sudo sh -c "echo 'memory_limit = $PHP_MEMORY_LIMIT' >> /etc/php/${{ matrix.php }}/cli/php.ini"
+          if [ -f /etc/php/${{ matrix.php }}/apache2/php.ini ]; then
+            sudo sh -c "echo 'memory_limit = $PHP_MEMORY_LIMIT' >> /etc/php/${{ matrix.php }}/apache2/php.ini"
+          fi
+
+          # Disable xdebug which greatly slow down unit testing
+          # Note: omitting xdebug from shivammathur/setup-php still results in xdebug being installed and enabled
+          if [ "${{ matrix.phpcoverage }}" == "true" ]; then
+            sudo sh -c "echo ';zend_extension=xdebug.so' > /etc/php/${{ matrix.php }}/mods-available/xdebug.ini"
+          fi
+          echo "PHP has been configured"
+
+      - name: Apt install additional requirements 
+        run: |
+          # apt install extra requirements as required
+          if [ "${{ matrix.endtoend }}" == "true" ]; then
+            sudo apt install -y software-properties-common
+            sudo add-apt-repository -y ppa:ondrej/php
+            sudo add-apt-repository -y ppa:ondrej/apache2
+            sudo apt update
+            sudo apt install -y libapache2-mod-php${{ matrix.php }}
+            # ubuntu-latest comes with a current version of google-chrome-stable and chromedriver
+          fi
+          if [[ "${{ github.repository }}" =~ /(spellcheck|recipe-authoring-tools)$ ]] || [ "${{ matrix.phpunit_suite }}" == "recipe-authoring-tools" ]; then
+            sudo apt install -y hunspell libhunspell-dev hunspell-en-us
+          fi
+
+      - name: Configure apache - endtoend test
+        if: ${{ matrix.endtoend == 'true' }}
+        run: |
+          # apache2 is installed and running by default in ubuntu
+          # update dir.conf to use index.php as the primary index doc
+          cat << EOF > __dir.conf
```

Comment:
Have updated for `dir.conf`

The other ones require the intermediate files because we're running a php script against them.  They probably could be "more inline" that we're not gaining much from the optimisation

Reply:
That syntax doesn't work, line break are not preserved.  I don't think we really need to be too concerned about minor syntax things like this

Reply:
Apologies, it turns out you need to quote the variable to preserve new lines.

```
DIR_CONF=$(cat << EOF
....
EOF)
sudo echo "$EOF" > /etc/apache2/mods-enabled/dir.conf
```

> I don't think we really need to be too concerned about minor syntax things like this

I think a decent portion of what a peer review is about is making the code clean. Adding extra files that we don't need is not clean.

Reply:
Please make this small change. If this syntax still doesn't work for some reason I'll concede it's not worth worrying about.